azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -0,0 +1,486 @@
+
+	if profile == nil {
+		return nil
+	}
+
+	servicePrincipalProfiles := &schema.Set{
+		F: resourceAzureRMKubernetesClusterServicePrincipalProfileHash,
+	}
+
+	values := make(map[string]interface{})
+
+	values["client_id"] = *profile.ClientID
+	if profile.Secret != nil {
+		values["client_secret"] = *profile.Secret
+	}
+
+	servicePrincipalProfiles.Add(values)
+
+	return servicePrincipalProfiles
+}
+
+func expandAzureRmKubernetesClusterLinuxProfile(d *schema.ResourceData) containerservice.LinuxProfile {
+	profiles := d.Get("linux_profile").([]interface{})
+	config := profiles[0].(map[string]interface{})
+
+	adminUsername := config["admin_username"].(string)
+
+	linuxKeys := config["ssh_key"].([]interface{})
+	sshPublicKeys := []containerservice.SSHPublicKey{}
+
+	key := linuxKeys[0].(map[string]interface{})
+	keyData := key["key_data"].(string)
+
+	sshPublicKey := containerservice.SSHPublicKey{
+		KeyData: &keyData,
+	}
+
+	sshPublicKeys = append(sshPublicKeys, sshPublicKey)
+
+	profile := containerservice.LinuxProfile{
+		AdminUsername: &adminUsername,
+		SSH: &containerservice.SSHConfiguration{
+			PublicKeys: &sshPublicKeys,
+		},
+	}
+
+	return profile
+}
+
+func expandAzureRmKubernetesClusterServicePrincipal(d *schema.ResourceData) *containerservice.ServicePrincipalProfile {
+	value, exists := d.GetOk("service_principal")
+	if !exists {
+		return nil
+	}
+
+	configs := value.(*schema.Set).List()
+
+	config := configs[0].(map[string]interface{})
+
+	clientId := config["client_id"].(string)
+	clientSecret := config["client_secret"].(string)
+
+	principal := containerservice.ServicePrincipalProfile{
+		ClientID: &clientId,
+		Secret:   &clientSecret,
+	}
+
+	return &principal
+}
+
+func expandAzureRmKubernetesClusterAgentProfiles(d *schema.ResourceData) []containerservice.AgentPoolProfile {
+	configs := d.Get("agent_pool_profile").([]interface{})
+	config := configs[0].(map[string]interface{})
+	profiles := make([]containerservice.AgentPoolProfile, 0, len(configs))
+
+	name := config["name"].(string)
+	count := int32(config["count"].(int))
+	dnsPrefix := config["dns_prefix"].(string)
+	vmSize := config["vm_size"].(string)
+	osDiskSizeGB := int32(config["os_disk_size_gb"].(int))
+	storageProfile := config["storage_profile"].(string)
+	vnetSubnetID := config["vnet_subnet_id"].(string)
+	osType := config["os_type"].(string)
+
+	profile := containerservice.AgentPoolProfile{
+		Name:           &name,
+		Count:          &count,
+		VMSize:         containerservice.VMSizeTypes(vmSize),
+		DNSPrefix:      &dnsPrefix,
+		OsDiskSizeGB:   &osDiskSizeGB,
+		StorageProfile: containerservice.StorageProfileTypes(storageProfile),
+		VnetSubnetID:   &vnetSubnetID,
+		OsType:         containerservice.OSType(osType),
+	}
+
+	profiles = append(profiles, profile)
+
+	return profiles
+}
+
+func kubernetesClusterStateRefreshFunc(client *ArmClient, resourceGroupName string, kubernetesClusterName string) resource.StateRefreshFunc {
```

Comment:
this appears unused?

Reply:
looks like we don't need this anymore after removing the poller